cpp/autosar/src/rules/A7-1-2/VariableMissingConstexpr.ql

```diff
@@ -62,5 +62,9 @@ where
   // Not assigned by a user in a constructor
   not exists(ConstructorFieldInit cfi | cfi.getTarget() = v and not cfi.isCompilerGenerated()) and
   // Ignore union members
-  not v.getDeclaringType() instanceof Union
+  not v.getDeclaringType() instanceof Union and
+  // Exclude variables in uninstantiated templates, as they may be incomplete
+  not v.isFromUninstantiatedTemplate(_) and
+  // Exclude compiler generated variables, which are not user controllable
+  not v.isCompilerGenerated()
```

Comment:
I think the alert stmt format could also be improved at the same time,

from the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide) - "Code elements should be placed in 'single quotes'"

I think var names count, we had done this previously in a few [other queries recently
](https://github.com/github/codeql-coding-standards/commit/c3a25d1e990b0e89323de1e1cace994304c66569)